webui/react/src/components/ThemeToggle.tsx

```diff
@@ -31,8 +31,8 @@ export const ThemeOptions: Record<Mode, ThemeOption> = {
 };
 
 const ThemeToggle: React.FC = () => {
-  const { ui } = useStore();
-  const storeDispatch = useStoreDispatch();
+  const ui = useUI();
+  const uiDispatch = useUIDispatch();
```

Comment:
Could I impose on you to make this more semantic? Something like:
```typescript
const [ui, toggleDarkMode] = useUI()
toggleDarkMode()
```
instead of asking a user to manually plumb data around different hooks?

Reply:
Hmm I think are more design decisions that will go into that compared to this could that be a separate pr/ticket?

Reply:
I think now where you're introducing a new API is the right place for that. What are your thoughts on a separate ticket?

Reply:
in terms of maybe wanting to take time to define what APIs we want to expose eg if we have a toggle do we want the explicit version of it or not, should we combine some of these etc

Reply:
Don't worry about getting it perfect on the first go. Use your judgement and it'll be an improvement over what it is now.